python/lsst/pipe/tasks/matchFakes.py

```diff
@@ -211,6 +221,28 @@ def composeFakeCat(self, fakeCats, skyMap):
 
         return pd.concat(outputCat)
 
+    def getVisitMatchedFakeCat(self, fakeCat, exposure):
+        """Trim the fakeCat to select particular field
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+                    The catalog of fake sources to add to the exposure
+        exposure : `lsst.afw.image.exposure.exposure.ExposureF`
+                    The exposure to add the fake sources to
+
+        Returns
+        -------
+        movingFakeCat : `pandas.DataFrame`
+            All fakes that satisfy conditions specified by matchExposureField
+        """
+        try:
+            selected = exposure.getInfo().getVisitInfo().getId() == fakeCat["visit"]
+        except Exception as e:
```

Comment:
We probably shouldn't be catching generic exceptions.

Reply:
Are there multiple typical exceptions or one main failure mode?

Reply:
Not really, the main one I was considering was if `visit` is not present in the `fakeCat`. I suppose I can just remove all these exceptions and let the task fail in case `visit` is not present in the catalog.